### PR TITLE
[Refactoring] Cleanup source code unit. Move instance methods into the best matching module

### PR DESCRIPTION
**Pull request by ketkarameya**
(You might have opinions related to this refactoring :P )

Our `impl` for `SourceCodeUnit` was getting too big and clunky . 
I split the `impl` block into multiple `impl` blocks. I move each `impl` block to the related module. 

I think it better organizes the code.  (Though its very _un-javaish_)
* in **mod** `constraint` we implement  the associated fn`is_satisfied` and `_is_satisfied` (semantics of constraint in some sense)
* in **mod** `edit` we implement  the associated fn `get_edit` and `get_edit_for_context` (semantics of edit in some sense)
* in **mod** `piranha_argument` we specify to implement the associated fn to implement semantics of piranha arguments like `dry_run`, `delete_file_if_empty`, `delete_consecutive_lines`, ... 

I feel it keeps the type and its semantics spatially close (making it easier to think, i guess, at least for me ).
Also this is kinda idiomatic in rust (according to few cyber warriors on reddit)

-----
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #355

### CODE REVIEW

File: src/lib.rs
Comment (ketkarameya) [inline src/lib.rs:88]: we were unnecessarily passing `piranha_arguments` (`SourceCodeUnit` already has `piranha_arguments`). So I updated this signature. 

This is the only signature I changed.

File: src/models/constraint.rs
Comment (lazaroclapp) [inline src/models/constraint.rs:46]: Ok, other than `pub(crate)`, this is the same code that was in `src_code_unit.rs` before, already under `impl SourceCodeUnit`, right? We are basically moving stuff from an object-oriented structure to an aspect-oriented structure (not exactly, but so to speak). Is my understanding here correct?

**Review by lazaroclapp — APPROVED** [review]
GitHub diffing UI could have made reviewing this a lot easier, but a couple copy pastes and manual runs of diff show this to be mostly code motion with just a few safe-looking changes 😅 

I am fine with the "trait"/"aspect" based organization here, even if it breaks class implementation across multiple modules.

File: src/models/constraint.rs
Comment (ketkarameya) [reply src/models/constraint.rs:46]: yes your understanding is correct. It is aspect oriented in some sense (Cross cutting and separation of concern). Actually that was kinda in my mind when i was doing this refactoring. Just wanted to make the main `source_code_unit` less crowded

**Comment by ketkarameya** [comment]
I will merge this stack after the i merge that larger stack

**Comment by ketkarameya** [comment]
Addressed the comment here - https://github.com/uber/piranha/pull/356/commits/6915f6b075a6331440d690817306cbd01d5d027d

**Pull request merged**
